### DISCUSSION TITLE
IntelliJ IDEA: accepting suggestion overwrites code

### DISCUSSION THREAD

**Discussion by randy-halim** [community/community#8039] [Copilot Conversations] [answered]
Sometimes, accepting a suggestion replaces the whole line with just that suggestion. For example:
```java
public class MalformedEntryException {
    //                              ^ cursor here
    // suggests `extends Exception`
    public MalformedEntryException() {
        // ...
    }
}
```
At this point, I accept it, and it ends up like this:
```java
extends Exception
    // ... constructor is fine
}
```
Also semi related: I lose IntelliJ's suggestions when Copilot suggests code. I have to hit Escape to get the suggestions back. It worked in the past before I updated. Using version `1.1.0`.

**Comment by jansorg** [comment, marked as answer]
@randy-halim Thanks! This will be fixed with the next update, which will be released soon.

Regarding IntelliJ's suggestions: This was a change to avoid mixing both IntelliJ's and Copilot's suggestions as long as you're typing what GitHub Copilot is showing in the editor.
Could you post a snippet where you'd expect IntelliJ's suggestions to show up? Would you expect them in addition or as a replacement of Copilot's suggestions?

Thanks!

**Reply by randy-halim** [reply]
Thank you! 
> Would you expect them in addition or as a replacement of Copilot's suggestions?

I was expecting IntelliJ's suggestions to be in addition to Copilot's suggestions; sometimes Copilot infers the wrong symbols/methods which I would have to dismiss and manually type the correct method. Example:
```java
public void saveToDb() {
  description.setDescriptionShort("My short description.");
  description.set
  //             ^ cursor
  // At this point, Copilot suggests `setDescriptionLong("My long description");`, which is great and all
  // but I don't have a method `setDescriptionLong`, I called it `setDescriptionFull`.
  // In the past, IntelliJ's suggestion would include `setDescriptionFull`
  // and I wouldn't need to type the full name manaually.
```
Hopefully that makes sense. Perhaps you could make an option where you can toggle IntelliJ's suggestions when Copilot suggests code? Thank you regardless!

**Reply by wjdp** [reply]
Ideally copilot would be able to use what IntelliJ knows about the code: what is the point of suggesting `setDescriptionLong` when it doesn't exist? Copilot could use the language server (or whatever IJ calls it) to improve its suggestions.
